sklearn/utils/tests/test_validation.py

```diff
@@ -631,35 +631,44 @@ def test_check_symmetric():
 
 
 def test_check_is_fitted():
-    # Check is ValueError raised when non estimator instance passed
-    assert_raises(ValueError, check_is_fitted, ARDRegression, "coef_")
-    assert_raises(TypeError, check_is_fitted, "SVR", "support_")
+    # Check is TypeError raised when non estimator instance passed
+    assert_raises(TypeError, check_is_fitted, ARDRegression)
+    assert_raises(TypeError, check_is_fitted, "SVR")
 
     ard = ARDRegression()
     svr = SVR()
 
     try:
-        assert_raises(NotFittedError, check_is_fitted, ard, "coef_")
-        assert_raises(NotFittedError, check_is_fitted, svr, "support_")
+        assert_raises(NotFittedError, check_is_fitted, ard)
+        assert_raises(NotFittedError, check_is_fitted, svr)
     except ValueError:
         assert False, "check_is_fitted failed with ValueError"
 
     # NotFittedError is a subclass of both ValueError and AttributeError
     try:
-        check_is_fitted(ard, "coef_", "Random message %(name)s, %(name)s")
+        check_is_fitted(ard, msg="Random message %(name)s, %(name)s")
     except ValueError as e:
         assert str(e) == "Random message ARDRegression, ARDRegression"
 
     try:
-        check_is_fitted(svr, "support_", "Another message %(name)s, %(name)s")
+        check_is_fitted(svr, msg="Another message %(name)s, %(name)s")
     except AttributeError as e:
         assert str(e) == "Another message SVR, SVR"
 
     ard.fit(*make_blobs())
     svr.fit(*make_blobs())
 
-    assert check_is_fitted(ard, "coef_") is None
-    assert check_is_fitted(svr, "support_") is None
+    assert check_is_fitted(ard) is None
+    assert check_is_fitted(svr) is None
+
+    assert_warns_message(
```

Comment:
```python
@pytest.mark.parametrize("params", [{'attributes': ['coefs_']}, {all_or_any=any}]
def test_check_is_fitted_deprecation(params):
    # FIXME: to be removed in 0.23
    warn_msg = 'Passing {} to check_is_fitted is deprecated'.format(list(params.keys())[0])
    with pytest.warns(DeprecationWarning, match=warn_msg):
        check_is_fitted(ard, **params)
```
It could be handy to have a separated test function to be removed next version.
We might use pytest (but the test will be removed anyway).

Reply:
Is it easier to remove a test than to remove the asserts? A comment might be nice but it will also just fail and so we won't forget ;)